web/src/engine/websites/decorators/Grouple.ts

```diff
@@ -0,0 +1,111 @@
+    'div#all img.img-responsive',
+    'div.text-center img[loading="lazy"]',
+].join(',');
+
+type ImagesData = {
+    url: string,
+    mirrors: string[];
+}
+
+type PageParameters = {
+    mirrors: string[],
+};
+
+/*************************************************
+ ******** Pages Extraction Methods ********
+ *************************************************/
+
+/**
+ * Extracts the pages from the HTML page of the given {@link chapter}.
+ * @param this - A reference to the {@link MangaScraper}
+ * @param chapter - A reference to the {@link Chapter} which contains the pages
+ * @param script - A script to extract the pages as \{ url : string, mirrors: string[] \}
+ */
+async function FetchPagesSinglePageJS(this: MangaScraper, chapter: Chapter, script: string = pagesWithServersScript, delay: number = 1500): Promise<Page[]> {
+    const uri = new URL(chapter.Identifier, this.URI);
+    uri.searchParams.set('mtr', '1');
+    const images = await FetchWindowScript<ImagesData[]>(new Request(uri), script, delay);
+    return images.map(image => new Page<PageParameters>(this, chapter, new URL(image.url), { mirrors: image.mirrors }));
+}
+/**
+ * A class decorator that adds the ability to extract all pages for a given chapter using the given JS {@link script}.
+ * The pages are extracted from the composed url based on the `Identifier` of the chapter and the `URI` of the website.
+ * @param script - A script to extract the pages as \{ url : string, mirrors: string[] \}
+ * @param delay - An initial delay [ms] before the {@link script} is executed
+ */
+export function PagesSinglePageJS(script = pagesWithServersScript, delay = 0) {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
+        Common.ThrowOnUnsupportedDecoratorContext(context);
+
+        return class extends ctor {
+            public async FetchPages(this: MangaScraper, chapter: Chapter): Promise<Page[]> {
+                return FetchPagesSinglePageJS.call(this, chapter, script, delay);
+            }
+        };
+    };
+}
+
+/**
+ * A class decorator that adds the ability to get the image data for a given page by loading the source asynchronous with the `Fetch API`.
+ * This method fetch pages by trying different urls. Page must have "mirrors: url,url, url, etc..." in Parameters.
+ */
+export function ImageAjaxWithMirrors() {
+    return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
+        Common.ThrowOnUnsupportedDecoratorContext(context);
+        return class extends ctor {
+            public async FetchImage(this: MangaScraper, page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+                return FetchMirroredImage.call(this, page, priority, signal);
+            }
+        };
+    };
+}
+
+async function FetchMirroredImage(this: MangaScraper, page: Page<PageParameters>, priority: Priority, signal: AbortSignal): Promise<Blob> {
```

Comment:
This method has a small footprint, is not exported and only used once.
Implementation might as well be merged directly into `PagesSinglePageJS`.

Reply:
You mean ImageAjaxWithMirrors but okay :)

Reply:
I cannot help with this error since i don't understand its cause